client/dialogs/dialog-reducer.ts

```diff
@@ -23,6 +23,18 @@ export default immerKeyedReducer(DEFAULT_DIALOG_HISTORY_STATE, {
   },
 
   ['@dialogs/close'](state, action) {
-    state.history.pop()
+    const { dialogType } = action.payload
+
+    if (dialogType === 'all') {
+      state.history = []
+      return
+    }
+
+    const dialogIndex = state.history.findIndex(h => h.type === dialogType)
+    if (dialogIndex < 0) {
+      return
+    }
+
+    state.history.splice(dialogIndex)
```

Comment:
No, I said when you close a dialog of a particular type, you close *all the dialogs up to that point*. What you've done here is *only close that particular dialog*. So if we have a settings dialog and it opens a "find your starcraft path dialog", then something happens that closes settings, your code here will leave the starcraft path dialog up despite its owner being closed. Seems not great, please implement as specified

Reply:
Oh, I thought the `Array.splice()` method would remove _all_ the elements from the given index to the end of array if the second argument is omitted. Or was that `Array.slice()`, I always confuse those two :d

Reply:
Yeah, `splice` is "remove some number of elements, optionally inserting new ones in their place". `slice` also wouldn't be what you want here (at least with a single argument), since that gives you a new array with the elements *starting* at that position.

Reply:
Wait, I'm a bit confused then. Isn't `splice` what we want then? If we have a history of [Dialog1, Dialog2, Dialog3] and something closes Dialog2, we'll be left with [Dialog1].

Reply:
No, you probably want to do `slice(0, index)`

Reply:
I'm still a bit confused I'm afraid. If we take the following hypothetical scenario:
- User opens the settings dialog
- User opens the "find your starcraft path" dialog within it

The `state.history` array should be [Dialog.Settings, Dialog.FindStarCraftPath]. Then if something closes the `Dialog.Settings`, `state.history.splice(index)` will basically be translated to `state.history.splice(0)`, which will remove all elements from the array and `state.history` should be equal to `[]`, which is what we want. I feel like I'm missing something rather obvious, but I'm just not seeing it atm ^^